draft-ietf-quic-qpack.md

```diff
@@ -340,20 +344,41 @@ MUST treat this as a connection error of type HTTP_QPACK_DECOMPRESSION_FAILED.
 
 ### State Synchronization
 
-The decoder instructions ({{decoder-instructions}}) signal key events at the
-decoder that permit the encoder to track the decoder's state.  These events are:
-
-- Complete processing of a header block
-- Abandonment of a stream which might have remaining header blocks
-- Receipt of new dynamic table entries
-
-Knowledge that a header block with references to the dynamic table has been
-processed permits the encoder to evict entries to which no unacknowledged
-references remain (see {{blocked-insertion}}).  When a stream is reset or
-abandoned, the indication that these header blocks will never be processed
-serves a similar function (see {{stream-cancellation}}).
-
-The decoder chooses when to emit Insert Count Increment instructions (see
+The decoder signals key events by emitting decoder instructions
+({{decoder-instructions}}) on the decoder stream.
+
+#### Completed Processing of a Header Block
+
+When the decoder finishes decoding a header block containing dynamic table
+references, it emits a Header Acknowledgement instruction
+({{header-acknowledgement}}).  The same Stream ID can be identified by multiple
+Header Acknowledgement instructions, as multiple header blocks can be sent on a
+single stream in the case of intermediate responses, trailers, and pushed
+requests.  Since frames carrying header blocks on each stream are received and
+processed in order, this gives the encoder precise feedback on which header
+blocks within a stream have been fully processed.
+
+#### Abandonment of a Stream
+
+When an endpoint receives a stream reset before the end of a stream or before
+all header blocks are processed on that stream, or when it abandons reading of a
+stream, it generates a Stream Cancellation instruction (see
+{{stream-cancellation}}).  This signals to the encoder that all references to
+the dynamic table on that stream are no longer outstanding.  A decoder with a
+maximum dynamic table capacity equal to zero (see
+{{maximum-dynamic-table-capacity}}) MAY omit sending Stream Cancellations,
+because the encoder cannot have any dynamic table references.  An encoder cannot
+infer from this instruction that any updates to the dynamic table have been
+received.
+
+The Header Acknowledgement and Stream Cancellation instructions permit the
+encoder to evict entries to which no unacknowledged references remain (see
+{{blocked-insertion}}).
```

Comment:
This wording feels weird.  Entries without unacknowledged references can always be evicted.  The instructions acknowledge the references themselves.